Add STEF exporter

tigrannajaryan opened the pull request:
Resolves https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/37352

This is the *First PR* done according to the requirements listed at https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-new-components

I intentionally left the Config barebones. It is possible we will add more config options in the future but I don't want to do it prematurely.

Subsequent PRs will add the implementation and will clarify how it works in the README.md.

tigrannajaryan committed:
Add STEF exporter

Resolves https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/37352

This is the *First PR* done according to the requirements listed at https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-new-components

I intentionally left the Config barebones. It is possible we will add more config
options in the future but I don't want to do it prematurely.

Subsequent PRs will add the implementation and will clarify how it works in the README.md.

dmitryax commented on exporter/stefexporter/metadata.yaml:
```diff
@@ -0,0 +1,13 @@
+  class: exporter
+  stability:
+    development: [metrics]
+  distributions: [contrib]
```
This should be empty until we publish it as part of https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-contrib
```suggestion
  distributions: []
```

tigrannajaryan replied:
Fixed.

dmitryax commented on exporter/stefexporter/factory.go:
```diff
@@ -0,0 +1,61 @@
+	exporter.Metrics, error,
+) {
+	cfg := config.(*Config)
+	exporterLogger := createLogger(set.TelemetrySettings.Logger)
```
We probably don't need this. `set.TelemetrySettings.Logger` is sampled by default. See default sampling config in https://github.com/open-telemetry/opentelemetry-collector/blob/main/service/telemetry/factory.go#L104-L109.

tigrannajaryan replied:
Fixed.

dmitryax reviewed (commented):
A couple of minor comments. Otherwise, LGTM

tigrannajaryan committed:
Fixes based on comments

tigrannajaryan committed:
Fix dependency versions

tigrannajaryan commented:
@dmitryax I addressed your comments.

The "Gen CODEOWNERS" github action [errors](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/12917590919/job/36024307388?pr=37400) with "2025/01/22 21:42:15 codeowners members duplicate in allowlist: jerrytfleung". Any idea what this is about? I can only see one "jerrytfleung" allowlist.txt, not sure what the duplicate is about. Unclear how my change is related to this error.

@atoulme since you are listed as a codeowner of githubgen, any idea what the error message is trying to tell me? Running `githubgen --skipgithub` locally does not give any errors (I don't have a PAT to try without `--skipgithub`).

atoulme commented:
@tigrannajaryan the issue reported is unrelated to your PR. A member of the allowlist has become an official member of OpenTelemetry and should be removed from the allowlist.
> 2025/01/22 21:42:15 codeowners members duplicate in allowlist: jerrytfleung
I will open a PR to fix. Thanks for reaching out.

tigrannajaryan commented:
Thanks @atoulme 
Please let me know once it is fixed, I will rebase.

atoulme commented:
You will not need to rebase for this, but I can trigger a rerun. The check-codeowners job runs by checking out the main branch of the repository. The fix is here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/37423

I will let you know of what this comes to in any case.

atoulme committed:
Merge branch 'main' into tigran/stefexporter

atoulme commented:
Alright, the `pull_request_target` is still pointing at the old main branch, so you'll need to rebase @tigrannajaryan. If you don't get around to it, I have updated with latest main here.

dmitryax committed:
make gendistributions

dmitryax committed:
make generate

tigrannajaryan commented:
Thanks @dmitryax @atoulme !
